src/Model/Entity/EnumTrait.php

```diff
@@ -10,11 +10,11 @@ trait EnumTrait {
 	 *
 	 * @link https://www.dereuromark.de/2010/06/24/static-enums-or-semihardcoded-attributes/
 	 * @param array|string|int|null $value Integer or array of keys or NULL for complete array result
-	 * @param array $options Options
+	 * @param \Traversable|array $options Options
```

Comment:
What's the need or reason?

Reply:
we found a fix for memory leak using yield

```
private static function __getSMSOptions()
    {
        yield [
            self::SMS_CONFIRMATION => __d('admin', 'Confirmation'),
            self::SMS_FORWARD => __d('admin', 'Reminder'),
            self::SMS_CHANGE => __d('admin', 'Change'),
            self::SMS_DELETE => __d('admin', 'Delete'),
            self::SMS_NOTIFICATION => __d('admin', 'Notification'),
        ];
    }

    /**
     * Returns (1) SMS template types array where the keys are template type ids
     * and the values are localized description or (2) the localized description
     * of one template type if $value param is provided.
     *
     * @param string $value Template type id.
     * @return string|string[]
     */
    public static function getSmsTemplateTypes($value = null)
    {
        $options = self::__getSMSOptions();
        return self::enum($value, $options);
    }
```